fix(message_handler): fix error when there is no record and fix crash

jrainville opened the pull request:
This was a little oopsie on my side.
I wasn't checking all the right conditions, but more importantly, I forgot to use an `else`. That caused a crash.

jrainville committed:
fix(message_handler): fix error when there is no record and fix crash

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 0a20351a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3074/1/) | 2023-01-11 17:46:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230111-174431-0a2035-pr3074.zip) |
| :heavy_check_mark: | 0a20351a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3074/1/) | 2023-01-11 17:48:53 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230111-174431-0a2035-pr3074.aar) |
| :heavy_check_mark: | 0a20351a | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3074/1/) | 2023-01-11 17:49:42 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230111-174431-0a2035-pr3074.zip) |
| :heavy_check_mark: | 0a20351a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3074/1/) | 2023-01-11 17:55:06 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3074/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 35a61351 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3074/3/) | 2023-01-12 16:18:24 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230112-161210-35a613-pr3074.zip) |
| :heavy_check_mark: | 35a61351 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3074/3/) | 2023-01-12 16:37:35 | ~17 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230112-162019-35a613-pr3074.zip) |
| :heavy_check_mark: | 35a61351 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3074/3/) | 2023-01-12 16:40:58 | ~21 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230112-161937-35a613-pr3074.aar) |
| :heavy_check_mark: | 35a61351 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3074/3/) | 2023-01-12 16:46:15 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3074/3//consoleText) |

jrainville committed:
feat(QuotedMessage): pass Deleted to QuotedMessage

This is to let the front end determine a new message if the quoted message is deleted instead of just unknown
